### PR TITLE
Automation: modal upgrades (no JIRA)

### PR DESCRIPTION
**Pull request by MattPutnam**
I noticed some issues when running tests locally. Sometimes the test wouldn't want to find a component inside of a modal. I also once had an issue where I inadvertently duplicated a test id (one on the page, one in the modal) and Selenium picked up the wrong one. So I upgraded Modal to a full Component so you can restrict the scope of test ids.

I also ended up modifying the structure of the library tests a bit, and added a utility for testing that modals can be closed (and re-opened).

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala [hunk 1]
```diff
@@ -1,61 +1,39 @@
...
-    }
-
     "Select options in the RP modal and generate breadcrumbs" in withWebDriver { implicit driver =>
       val user = UserPool.chooseAnyUser
 
       withSignIn(user) { _ =>
         val page = new DataLibraryPage().open
 
-        page.openResearchPurposeModal()
+        testModal(page.openResearchPurposeModal())
```
Comment (aednichols) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataUseAwareSearchSpec.scala:24]: Nice.

**Review by aednichols — APPROVED** [review]
LGTM, I like the big-picture thinking.

Otherwise... just wait for tests to pass?

**Review by davidangb — APPROVED** [review]
:+1: this looks right, and I love the refactoring!

On the flip side, we introduce a UI regression frequently when we tweak the modals. Could you make sure to work with Andrew to ensure we've at least manually tested the various test+corner cases?
